File: 2021-11/2021-11-14 677.键值映射/MapSum.cs
using System.Collections.Generic;
using System.Linq;

public class MapSum
{
    public Dictionary<string, int> Dict { get; set; } = new Dictionary<string, int>();

    public void Insert(string key, int val) => Dict[key] = val;

    public int Sum(string prefix) =>
        Dict.Where(x => x.Key.StartsWith(prefix))
            .Sum(x => x.Value);
}

/**
 * Your MapSum object will be instantiated and called as such:
 * MapSum obj = new MapSum();
 * obj.Insert(key,val);
 * int param_2 = obj.Sum(prefix);
 */

File: 2021/2021-11/2021-11-14 677.键值映射/MapSum.go
package main

import "strings"

func main() {

}

type MapSum struct {
	hash map[string]int
}

func Constructor() MapSum {
	return MapSum{
		hash: make(map[string]int, 0),
	}
}

func (m *MapSum) Insert(key string, val int) {
	m.hash[key] = val
}

func (m *MapSum) Sum(prefix string) int {
	sum := 0
	for key, val := range m.hash {
		if strings.HasPrefix(key, prefix) {
			sum += val
		}
	}
	return sum
}

/**
 * Your MapSum object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Insert(key,val);
 * param_2 := obj.Sum(prefix);
 */
